### DISCUSSION TITLE
Can next-auth lazy load provider on page refresh?

### DISCUSSION THREAD

**Discussion by qhuynhvhslab** [nextauthjs/next-auth#7383] [Help]
### Question 💬

Hi,

When we use multiple providers in [...nextauth] file, my working place has a requirement to load the secret through a database, something like this

```
 TwitterProvider({
      clientId: await getVariableConfig('twitterId'),
      clientSecret: await getVariableConfig('twitterSecret'),
      version: '2.0',
    })
```

So I notice each time F5, it hits the db twice to get the credentials. I know normally we tend to use the environment variables but in this case if we still have to hit database to query, is there any other way to not request on refresh? Except caching which we dont have it yet

### How to reproduce ☕️

Instead of configure the secret to be environment variable, just query it from the database and refresh the page.

we use SSR to query nextauth session on all protected route

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by balazsorban44** [comment]
Yes. Check out https://next-auth.js.org/configuration/initialization#advanced-initialization

Then you can read the DB once before you pass it to `NextAuth`
